src/core/Akka.TestKit/EventFilter/Internal/EventFilterApplier.cs

```diff
@@ -442,7 +442,23 @@ public EventFilterFactory And
                 ? TestKitExtension.For(system).TestEventFilterLeeway
                 : _testkit.TestKitSettings.TestEventFilterLeeway;
 
-            var timeoutValue = timeout.HasValue ? _testkit.Dilated(timeout.Value) : leeway;
+            TimeSpan timeoutValue;
+            if (timeout.HasValue)
+            {
+                timeoutValue = _testkit.Dilated(timeout.Value);
+            }
+            else
+            {
+                timeoutValue = _testkit.RemainingOrDilated(leeway);
```

Comment:
Check if we're inside a Within block, if we do, use Remaining, else use a dilated leeway